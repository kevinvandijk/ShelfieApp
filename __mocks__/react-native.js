const rn = require('react-native');

jest.mock('Linking', () => {
  return {
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    openURL: jest.fn(),
    canOpenURL: jest.fn(),
    getInitialURL: jest.fn(),
  };
});

jest.mock('NetInfo', () => {
  return {
    isConnected: {
      fetch: () => {
        return new Promise((accept) => {
          accept(true);
        });
      }
    }
  };
});

jest.mock('AsyncStorage', () => {
  function getTestData() {
    return {};
  }

  function multiGetTestData() {
    return [
      ['key1', JSON.stringify({ valor: 1 })],
      ['key2', JSON.stringify({ valor: 2 })],
    ];
  }

  return {
    setItem: jest.fn(() => {
      return new Promise((resolve) => {
        resolve(null);
      });
    }),

    multiSet: jest.fn(() => {
      return new Promise((resolve) => {
        resolve(null);
      });
    }),

    getItem: jest.fn(() => {
      return new Promise((resolve) => {
        resolve(JSON.stringify(getTestData()));
      });
    }),

    multiGet: jest.fn(() => {
      return new Promise((resolve) => {
        resolve(multiGetTestData());
      });
    }),

    removeItem: jest.fn(() => {
      return new Promise((resolve) => {
        resolve(null);
      });
    }),

    getAllKeys: jest.fn(() => {
      return new Promise((resolve) => {
        resolve([]);
      });
    }),

    flushGetRequests: jest.fn(() => {
      return new Promise((resolve) => {
        resolve();
      });
    })
  };
});

module.exports = rn;
