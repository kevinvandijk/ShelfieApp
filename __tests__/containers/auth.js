import 'react-native';
import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import AuthContainer from '../../app/containers/auth';
import configureStore from '../../app/store/configure-store';

jest.mock('../../app/lib/i18n');

describe('<AuthContainer>', () => {
  it('renders', () => {
    renderer.create(
      <Provider store={ configureStore() }>
        <AuthContainer />
      </Provider>
    );
  });
});
