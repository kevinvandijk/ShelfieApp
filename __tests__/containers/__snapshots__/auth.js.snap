exports[`<AuthContainer> renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }>
  <View
    style={
      Object {
        "alignItems": "stretch",
        "backgroundColor": "#fff",
        "flexDirection": "row",
        "height": 44,
      }
    }>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderBottomColor": "#E24741",
          "borderBottomWidth": 3,
          "flex": 1,
          "justifyContent": "center",
          "opacity": 1,
        }
      }
      testID={undefined}>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#DDD",
              "fontFamily": "Lato-Black",
              "fontSize": 11,
            },
            Object {
              "color": "#515253",
            },
          ]
        }>
        TABS.LOGIN
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderBottomColor": "#DDD",
          "borderBottomWidth": 0.5,
          "flex": 1,
          "justifyContent": "center",
          "opacity": 1,
        }
      }
      testID={undefined}>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#DDD",
              "fontFamily": "Lato-Black",
              "fontSize": 11,
            },
            null,
          ]
        }>
        TABS.REGISTER
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }>
    <View
      style={
        Array [
          Object {
            "alignItems": "stretch",
            "backgroundColor": "#eee",
            "flex": 1,
            "justifyContent": "flex-end",
            "overflow": "hidden",
            "paddingHorizontal": 15,
            "paddingVertical": 15,
          },
        ]
      }>
      <View
        style={
          Object {
            "backgroundColor": "#fff",
            "flex": 1,
            "justifyContent": "space-between",
            "marginBottom": 15,
            "overflow": "hidden",
            "paddingHorizontal": 12.5,
            "paddingVertical": 12.5,
          }
        }>
        <View>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#515253",
                  "fontFamily": "Lato-Regular",
                  "fontSize": 12,
                  "lineHeight": 16.5,
                },
                Object {
                  "color": "#9B9B9B",
                  "marginBottom": 43.5,
                  "marginLeft": 26.5,
                  "marginRight": 26.5,
                  "textAlign": "center",
                },
              ]
            }>
            content.welcome
          </Text>
        </View>
        <View>
          <TextInput
            autoCapitalize="none"
            autoCorrect={false}
            blurOnSubmit={false}
            enablesReturnKeyAutomatically={true}
            keyboardType="email-address"
            onChangeText={[Function]}
            onSubmitEditing={[Function]}
            placeholder="content.email"
            returnKeyType="next"
            style={
              Array [
                Object {
                  "alignSelf": "stretch",
                  "backgroundColor": "#eee",
                  "color": "#303030",
                  "fontFamily": "Lato-Regular",
                  "fontSize": 12,
                  "height": 44,
                  "lineHeight": 44,
                  "padding": 15,
                },
                undefined,
              ]
            }
            value="" />
          <TextInput
            enablesReturnKeyAutomatically={true}
            onChangeText={[Function]}
            onSubmitEditing={[Function]}
            placeholder="content.password"
            returnKeyType="go"
            secureTextEntry={true}
            style={
              Array [
                Object {
                  "alignSelf": "stretch",
                  "backgroundColor": "#eee",
                  "color": "#303030",
                  "fontFamily": "Lato-Regular",
                  "fontSize": 12,
                  "height": 44,
                  "lineHeight": 44,
                  "padding": 15,
                },
                Object {
                  "marginTop": 15,
                },
              ]
            }
            value="" />
        </View>
      </View>
      <View
        style={undefined}>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#E24546",
              "height": 44,
              "justifyContent": "center",
              "marginBottom": 15,
              "opacity": 1,
            }
          }
          testID={undefined}>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#fff",
                  "fontSize": 12,
                },
                undefined,
                null,
              ]
            }>
            CONTENT.LOGIN
          </Text>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#3B5998",
              "height": 44,
              "justifyContent": "center",
              "opacity": 1,
            }
          }
          testID={undefined}>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#fff",
                  "fontSize": 12,
                },
                undefined,
                null,
              ]
            }>
            CONTENT.FACEBOOKLOGIN
          </Text>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "right": 0,
            },
            Object {
              "height": 0,
            },
            undefined,
          ]
        } />
    </View>
  </View>
</View>
`;
